subprojects/docs/src/snippets/customPlugins/customPluginWithConvention/groovy/build.gradle

```diff
@@ -1,15 +1,12 @@
-abstract class GreetingPluginExtension {
-    abstract Property<String> getMessage()
-
-    GreetingPluginExtension() {
-        message.convention('Hello from GreetingPlugin')
-    }
+interface GreetingPluginExtension {
+    Property<String> getMessage()
 }
 
 class GreetingPlugin implements Plugin<Project> {
     void apply(Project project) {
         // Add the 'greeting' extension object
         def extension = project.extensions.create('greeting', GreetingPluginExtension)
+        extension.message.convention('Hello from GreetingPlugin')
```

Comment:
Should we explicitly comment that this line sets the extension message? Or do we think that's clear enough here that we can omit it? (same in the kotlin example)